### PR TITLE
15538 Don't set initial restoration type + misc cleanup

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#15538

*Description of changes:*
- app version = 7.3.24
- misc package updates
- refactored restoration filing init (to support null type)
- refactored restoration filing resume
- deleted unneeded application types

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-693-v4u2a128.web.app

**Comment by severinbeauvais** [comment]
Sample initial draft filing (type=null):

image

Sample initial restoration application (no initial type):

image

**Comment by severinbeauvais** [comment]
@hfekete I added you as reviewer since this is Filings UI code and you might want to incorporate these changes in the new business dashboard.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-693-v4u2a128.web.app

File: src/mixins/filing-mixin.ts [hunk 1]
```diff
@@ -155,9 +155,10 @@ export default class FilingMixin extends DateMixin {
 
   /**
    * Builds a Restoration filing body.
+   * @param restorationType The type of restoration. May be null.
    * @returns the filing body
    */
-  buildRestorationFiling (restorationType = FilingSubTypes.FULL_RESTORATION): RestorationFilingIF {
+  buildRestorationFiling (restorationType: FilingSubTypes): RestorationFilingIF {
```
Comment (severinbeauvais) [inline src/mixins/filing-mixin.ts:161]: Just use what's passed in -- don't fall back to Full Restoration.

**Review by JazzarKarim — APPROVED** [review]
LGTM 👍

File: src/components/Dashboard/StaffNotation.vue [hunk 2]
```diff
@@ -386,13 +381,15 @@ export default class StaffNotation extends Mixins(AllowableActionsMixin, FilingM
...
-        url = `${this.getEditUrl}${this.getIdentifier}/` + restorationType + `?restoration-id=${id}`
+      // navigate to Edit UI for limited restoration extension filing
+      // navigate to Edit UI for limited restoration to full filing
+      // navigate to Create UI for full or limited restoration filing
+      if (restorationType === FilingSubTypes.LIMITED_RESTORATION_EXTENSION) {
+        url = `${this.getEditUrl}${this.getIdentifier}/limitedRestorationExtension?restoration-id=${id}`
+      } else if (restorationType === FilingSubTypes.LIMITED_RESTORATION_TO_FULL) {
+        url = `${this.getEditUrl}${this.getIdentifier}/limitedRestorationToFull?restoration-id=${id}`
+      } else {
+        url = `${this.getCreateUrl}restoration-business-name?id=${this.getIdentifier}`
```
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:392]: I've added this direct route to step 1 of the restoration filing, which saves a default re-route and then another re-route for restorations.

Ditto in TodoList.

And also I updated the unit tests.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-693-v4u2a128.web.app

File: src/components/Dashboard/StaffNotation.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/Dashboard/StaffNotation.vue:392]: @hfekete Here's why a route is specified for Create UI ^^.

Comment (severinbeauvais) [reply src/components/Dashboard/StaffNotation.vue:392]: @patrickpeinanw FYI, and also see other changes in this PR. Thanks!

**Pull request merged**